core/chains/evm/logpoller/log_poller.go

```diff
@@ -153,10 +153,15 @@ func NewLogPoller(orm ORM, ec Client, lggr logger.Logger, pollPeriod time.Durati
 }
 
 type Filter struct {
-	Name      string // see FilterName(id, args) below
-	EventSigs evmtypes.HashArray
-	Addresses evmtypes.AddressArray
-	Retention time.Duration
+	Name         string // see FilterName(id, args) below
+	Addresses    evmtypes.AddressArray
+	EventSigs    evmtypes.HashArray // list of possible values for eventsig (aka topic1)
+	Topic2       evmtypes.HashArray // list of possible values for topic2
+	Topic3       evmtypes.HashArray // list of possible values for topic3
+	Topic4       evmtypes.HashArray // list of possible values for topic4
+	Retention    time.Duration      // maximum amount of time to retain logs
+	MaxLogsKept  *ubig.Big          // maximum number of logs to retain
```

Comment:
I believe this is gonna be used during PruneOldLogs, right? Couting during delete might be heavy, have you thought about the delete query that would respect that value?

Reply:
I haven't thought about that yet, but I suspect we can find a way to make it efficient. If pruning all filters at once is a problem, one thing we could do is do one filter at a time... so that only the rows for a single filter are locked at once while we count & prune them.